napari/_qt/qt_main_window.py

```diff
@@ -818,6 +822,17 @@ def _setup_npe1_samples_menu(self):
         plugin_manager.events.unregistered.connect(_rebuild_npe1_samples_menu)
         _rebuild_npe1_samples_menu()
 
+    def _setup_npe1_plugins_menu(self):
+        from napari.plugins._npe2 import index_npe1_adapters
+
+        index_npe1_adapters()
```

Comment:
@lucyleeow code looks fine in general (especially since it's slated for removal) but this call should be moved out of this function into a function that is guaranteed to run on `viewer` init. Otherwise, the npe1 shim won't work - and we want to keep shimmed plugins even after we deprecate npe1.

Reply:
I've moved it to above `_add_menus` in `Window.__init__` because it did not seem to fit with the menu building inside `_add_menus`. Happy to amend though.